### PR TITLE
[ANALYTICS-1724] Add Unit tests

### PR DESCRIPTION
**Pull request by JeremyKulchyk**
## Summary of Changes
Added unit tests for all files, and fixed a couple bugs on the way - current coverage is 98%.

## Housekeeping Checklist
- [x] Linked the PR to a Jira ticket?
- [x] Linked the Jira ticket to the PR?
- [x] Checked Draft/vs Ready to Review?
- [x] Tagged the reviewers if Ready for Review?

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1,8 +1,295 @@
...
 @pytest.fixture
 def emission_composition():
-    return EmissionComposition()
+    return EmissionComposition(
+        co2=float(random.randint(0, 10000)),
+        co=float(random.randint(0, 10000)),
+        c1=float(random.randint(0, 10000)),
+        vocs=float(random.randint(0, 10000)),
+        n2o=float(random.randint(0, 10000)),
+    )
```
Comment (RobBlumberg) [inline tests/conftest.py:57]: if were gonna use `random` throughout to generate values, would be good to set a seed at the top of the module. Could become difficult to track down failing test with non reproducible values, and generating random values may cause us to hit unexpected edge cases at some pt

File: tests/test_emission_composition.py [hunk 2]
```diff
@@ -1,8 +1,42 @@
...
+    assert hasattr(emission_composition, "c1")
+    assert hasattr(emission_composition, "co")
+    assert hasattr(emission_composition, "co2")
+    assert hasattr(emission_composition, "vocs")
+    assert hasattr(emission_composition, "n2o")
+
+
+def test_emission_composition_add(emission_composition):
+
+    assert isinstance(emission_composition + emission_composition, EmissionComposition)
```
Comment (RobBlumberg) [inline tests/test_emission_composition.py:17]: thinking it might be nice to have another test that checks that the `__add__` logic is working as expected, like something along the lines of
```python
-> manual_result = <do manually addition>
assert emission_composition + emission_composition == manual_result
```
and could do similar for addition of emission profiles

File: tests/test_emissions_profile.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+def test_emissions_profile_total(emissions_profile):
+    assert isinstance(emissions_profile._total, EmissionComposition)
+
+
+def test_emissions_profile_allocate(emissions_profile):
+    print(emissions_profile.vent.co2)
+    allocation = emissions_profile.allocate(portion=0.5)
+    print(emissions_profile.vent.co2)
+    print(allocation.vent.co2)
```
Comment (RobBlumberg) [inline tests/test_emissions_profile.py:29]: assuming you forgot to get rid of these debug `prints`?

**Review by RobBlumberg — COMMENTED** [review]
Test suite looking great with really great coverage as well ! 

Screen Shot 2022-03-03 at 10 08 04 PM

Just left a few suggestions. Also, would be good to structure the test modules in `tests/` to mimic the structure in `emissions_framework/` (e.g like in [va_framework/tests](https://github.com/ValidereInc/va_framework/tree/master/tests))

File: tests/conftest.py [hunk 4]
```diff
@@ -1,8 +1,295 @@
...
+    return {"emissions_record_1": EmissionsProfile()}
+
+
+# Emissions record 2
+@pytest.fixture
+def emissions_record_2():
+    return {"emissions_record_2": EmissionsProfile()}
+
+
+# Node
```
[the remark this answers is not in the document]
Comment (JeremyKulchyk) [reply tests/conftest.py:107]: addressed

File: tests/test_emissions_profile.py (hunk 3 above)
Comment (JeremyKulchyk) [reply tests/test_emissions_profile.py:29]: addressed

File: tests/conftest.py (hunk 1 above)
Comment (JeremyKulchyk) [reply tests/conftest.py:57]: Addressed

File: tests/test_emission_composition.py (hunk 2 above)
Comment (JeremyKulchyk) [reply tests/test_emission_composition.py:17]: Addressed

File: tests/conftest.py [hunk 5]
```diff
@@ -1,8 +1,194 @@
...
+            edge_id=edge_id,
+            product=product,
+            from_node=from_node,
+            to_node=to_node,
+            product_type="oil",
+            emissions_generator=emissions_generator,
+            emissions_generator_params={"fuel__distance": distance},
+        )
+
+    return create_edge
```
Comment (RobBlumberg) [inline tests/conftest.py:147]: fixture factories looking nice !

**Review by RobBlumberg — APPROVED** [review]
LGTM
